### PR TITLE
[doc] Add CUDA description to package

### LINKED ISSUE #114364: Improve usability of CUDA package by adding description of CUDA
### 📚 The doc issue

While reading through the documentation, I came to the [Torch CUDA page](https://pytorch.org/docs/stable/cuda.html#). The description explained that the package adds support for CUDA but didn't explain what CUDA is. It would be helpful to new users reading the docs to be given some background on CUDA as well.

### Suggest a potential alternative/fix

My suggestion is to add CUDA context to [Torch CUDA page](https://pytorch.org/docs/stable/cuda.html#). Something like the following text:

```
CUDA is a parallel computing platform and programming model developed by NVIDIA for general computing on GPUs.
With CUDA, developers are able to dramatically speed up computing applications by harnessing the power of GPUs.
See https://developer.nvidia.com/cuda-zone for more details.
```

I am willing to push the PR for this change.

### PR DESCRIPTION
**Pull request by hickeyma**
Fixes #114364

### CODE REVIEW

File: torch/cuda/__init__.py [hunk 1]
```diff
@@ -1,6 +1,12 @@
 r"""
 This package adds support for CUDA tensor types.
 
+CUDA is a parallel computing platform and programming model
+developed by NVIDIA for general computing on GPUs.
+With CUDA, developers are able to dramatically speed up computing
+applications by harnessing the power of GPUs.
+See https://developer.nvidia.com/cuda-zone for more details.
```
Comment (malfet) [inline torch/cuda/__init__.py:8]: Would https://en.wikipedia.org/wiki/CUDA be a better link here?

**Review by malfet — COMMENTED** [review]
To be frank I'm a bit on the edge here: on one hand it feels good to add a bit more explanation, but on the other how to define when one should stop?
Should we define what GPU is? Or thread? Or half precision data type?

File: torch/cuda/__init__.py [hunk 2]
```diff
@@ -1,6 +1,12 @@
 r"""
 This package adds support for CUDA tensor types.
 
+CUDA is a parallel computing platform and programming model
+developed by NVIDIA for general computing on GPUs.
+With CUDA, developers are able to dramatically speed up computing
+applications by harnessing the power of GPUs.
```
[the remark this answers is not in the document]
Comment (hickeyma) [reply torch/cuda/__init__.py:7]: Agree, removed.

File: torch/cuda/__init__.py (hunk 1 above)
Comment (hickeyma) [reply torch/cuda/__init__.py:8]: I think so, provides more details and context.

**Comment by hickeyma** [comment]
> To be frank I'm a bit on the edge here: on one hand it feels good to add a bit more explanation, but on the other how to define when one should stop?

@malfet That's a reasonable point. Adding some small context and link to details on wikipedia seems like a good balance.

File: torch/cuda/__init__.py (hunk 1 above)
Comment (hickeyma) [reply torch/cuda/__init__.py]: Added

**Comment by hickeyma** [comment]
Thanks for the review @malfet. Comments updated and ready for review again.

**Comment by hickeyma** [comment]
Ping @malfet @svekars @carljparker

**Pull request closed without merging**